package/MDAnalysis/coordinates/DCD.py

```diff
@@ -89,438 +69,214 @@
-        np.put(unitcell, _ts_order, np.concatenate([lengths, angles]))
-        return unitcell
+            self.convert_pos_from_native(self.ts.dimensions[:3])
 
     def close(self):
-        """Close trajectory and flush buffers."""
-        if hasattr(self, 'dcdfile') and self.dcdfile is not None:
-            self._finish_dcd_write()
-            self.dcdfile.close()
-            self.dcdfile = None
-
+        """close reader"""
+        self._file.close()
 
-class DCDReader(base.ReaderBase):
-    """Reads from a DCD file
-
-    :Data:
-        ts
-          :class:`Timestep` object containing coordinates of current frame
-
-    :Methods:
-        ``dcd = DCD(dcdfilename)``
-           open dcd file and read header
-        ``len(dcd)``
-           return number of frames in dcd
-        ``for ts in dcd:``
-           iterate through trajectory
-        ``for ts in dcd[start:stop:skip]:``
-           iterate through a trajectory
-        ``dcd[i]``
-           random access into the trajectory (i corresponds to frame number)
-        ``data = dcd.timeseries(...)``
-           retrieve a subset of coordinate information for a group of atoms
-
-    Note
-    ----
-    The DCD file format is not well defined. In particular, NAMD and CHARMM use
-    it differently. Currently, MDAnalysis tries to guess the correct format for
-    the unitcell representation but it can be wrong. **Check the unitcell
-    dimensions**, especially for triclinic unitcells (see `Issue 187`_ and
-    :attr:`Timestep.dimensions`). A second potential issue are the units of
-    time (TODO).
-
-
-    .. versionchanged:: 0.9.0
-       The underlying DCD reader (written in C and derived from the
-       catdcd/molfile plugin code of VMD) is now reading the unitcell in
-       NAMD ordering: ``[A, B, C, sin(gamma), sin(beta),
-       sin(alpha)]``. See `Issue 187`_ for further details.
-
-    .. _Issue 187: https://github.com/MDAnalysis/mdanalysis/issues/187
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return len(self._file)
 
-    .. versionchanged:: 0.11.0
-       Frames now 0-based instead of 1-based.
-       Native frame number read into `ts._frame`.
-       Removed `skip` keyword and functionality.
+    def _reopen(self):
+        """reopen trajectory"""
+        self.ts.frame = 0
+        self._frame = -1
+        self._file.close()
+        self._file.open('r')
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self._frame = i - 1
+        self._file.seek(i)
+        return self._read_next_timestep()
 
-    """
-    format = 'DCD'
-    flavor = 'CHARMM'
-    units = {'time': 'AKMA', 'length': 'Angstrom'}
-    _Timestep = Timestep
+    def _read_next_timestep(self, ts=None):
+        """copy next frame into timestep"""
+        if self._frame == self.n_frames - 1:
+            raise IOError('trying to go over trajectory limit')
```

Comment:
Currently consistency with other readers is why I would prefer a IOError. Once this is merged I can look into standardizing the behavior.